File: prototipo-tp-introduccion/src/client/src/logger/logger.go
package logger

import (
	"fmt"
	"log/slog"
)

type LogResult string

const (
	InProgress LogResult = "in-progress"
	Fail       LogResult = "fail"
	Success    LogResult = "success"
)

func buildMessage(action string, result LogResult) string {
	return fmt.Sprintf("action=%s result=%s", action, result)
}

func Info(action string, result LogResult, args ...any) {
	message := buildMessage(action, result)
	slog.Info(message, args...)
}

func Warn(action string, result LogResult, args ...any) {
	message := buildMessage(action, result)
	slog.Warn(message, args...)
}

func Error(action string, result LogResult, args ...any) {
	message := buildMessage(action, result)
	slog.Error(message, args...)
}

File: prototipo-tp-introduccion/src/server/src/logger/logger.py
import logging
from enum import StrEnum

class LogResult(StrEnum):
    in_progress = "in-progress"
    fail = "fail"
    success = "success"

def init():
	logging.basicConfig(
		level=logging.INFO,
		format="%(asctime)s %(levelname)s %(message)s",
		datefmt="%Y/%m/%d %H:%M:%S",
	)

def _parse_args(args) -> str:
	parsed_args = []
	for pair in zip(args[::2], args[1::2]):
		parsed_args.append(f"{pair[0]}={pair[1]}")
	return " ".join(parsed_args)

def info(action: str, result: LogResult, *args):
	logging.info(f"action={action} result={result} {_parse_args(args)}")

def error(action: str, result: LogResult, *args):
	logging.error(f"action={action} result={result} {_parse_args(args)}")
